Add flag to allow completed steps to be skipped

K-Meech opened the pull request:
Closes https://github.com/UCL-ARC/Enigma-PD-WML/issues/14

Main changes:
- adds an optional `-o` flag (for 'overwrite'). When provided, this will ensure all pipeline steps are re-run + previous output is overwritten. When not provided, the pipeline will try to re-use existing output files, skipping steps that are already complete. This is useful when the pipeline fails at a late stage and you don't want to re-run time consuming steps.

- To enable this, the main part of the pipeline is now split into steps (each its own function): `fslAnat`, `flairPrep`, `ventDistMapping`, `prepImagesForUnet`, `unetsPgs`, `processOutputs`. Each will check if it's output files already exist + skip later steps if overwrite is false.

- The setup steps are now also wrapped into a function `setupRunAnalysis`, mostly to ensure all output is captured in the log file without having to include `>> $overall_log 2>&1` on multiple lines

- Fixed a filename typo (checked directly with Hamied): `flairvol2t1brain.nii.gz`

drmatthews reviewed (changes requested):
Nice - looks a lot tidier! The only thing I'd suggest is adding some more documentation to the README.md file to describe the script (or at least that the pipeline is actually this set of shell script functions). At the least the new `-O` flag should be documented.

K-Meech commented:
Thanks @drmatthews ! I've now added documentation for the `-o` option. Also, while looking through the readme, I realised I'd been using 'sample' instead of 'subject' for the log names - this is also fixed now. 

As for adding more description of the script itself, I'm currently working on that in a separate branch as part of https://github.com/UCL-ARC/Enigma-PD-WML/issues/3 . Would it be fine to include that in a separate PR along with various other documentation improvements?

drmatthews reviewed (approved):
👍
